### PR TITLE
hideUnderline

### PR DESCRIPTION
**Pull request by dimmetrius**
Whether to hide the line under the input field.

### CODE REVIEW

**Comment by dimmetrius** [comment]
I signed it!

**Comment by Hixie** [comment]
Can you elaborate on the use case?

We should consider whether to implement this more like DropdownButtonHideUnderline. Depending on the use case, that might make more sense.

File: packages/flutter/lib/src/material/input_decorator.dart [hunk 1]
```diff
@@ -416,7 +429,7 @@ class InputDecorator extends StatelessWidget {
       decoration: new BoxDecoration(
         border: new Border(
           bottom: new BorderSide(
-            color: borderColor,
+            color: hideUnderline ? Colors.transparent : borderColor,
```
Comment (Hixie) [inline packages/flutter/lib/src/material/input_decorator.dart]: setting the style to BorderStyle.none would be more efficient.

Comment (dimmetrius) [inline packages/flutter/lib/src/material/input_decorator.dart]: ok i'll use BorderStyle.none for hide underline. in my project i need Text Input with placeholder and custom underline

**Comment by dimmetrius** [comment]
I signed it!

**Comment by Hixie** [comment]
(Assuming that the discussion of use cases above leads to the conclusion that this is the right approach, this will need tests.)

**Comment by dimmetrius** [comment]
ok i'll write tests

**Comment by Hixie** [comment]
Well before you write tests the most important thing is to describe why you need this at all so that we can decide if this is the right approach.

**Comment by Hixie** [comment]
@dimmetrius Will you be able to continue working on this? It's totally fine if not; we can take it over (though it might take longer). I'm just asking because I'm trying to clean out our review queue.

**Comment by abarth** [comment]
There's also `InputDecoration.collapsed`, which handles many of these use cases.

**Comment by Hixie** [comment]
@dimmetrius I'm going to close this PR for now, but please don't hesitate to reopen it if you have a chance to work on it again. We'll get to it eventually if you don't. Thanks for the contribution so far!

**Comment by jslavitz** [comment]
@Hixie Seems like this is an aesthetic feature?

**Comment by Hixie** [comment]
Yeah I never really understood what @dimmetrius exactly wanted to do here. Without knowing the use case, it's possible that we should just close this without landing.

**Pull request closed without merging**